python/src/equistore/operations/reduce_over_samples.py

```diff
@@ -101,69 +101,94 @@ def _reduce_over_samples_block(
     return result_block
 
 
-def sum_over_samples(tensor: TensorMap, sample_names: List[str]) -> TensorMap:
+def _reduce_over_samples(
+    tensor: TensorMap, samples_names: Union[List[str], str], reduction: str
+) -> TensorMap:
     """Create a new :py:class:`TensorMap` with the same keys as
     as the input ``tensor``, and each :py:class:`TensorBlock` is obtained
-    summing the corresponding input :py:class:`TensorBlock` over the rows with
-    the same ``sample_names``.
+    summing the corresponding input :py:class:`TensorBlock` over the ``samples_names``
+    indeces.
```

Comment:
```suggestion
    indices.
```

Here and everywhere else =)